### PR TITLE
[ci] remove bazel logs only on mac

### PR DESCRIPTION
**Pull request by can-anyscale**
As per comment on the code; remove bazel logs only on mac. Other platform runs in docker so it doesn't matter if the log is removed.

Test:
- CI

### CODE REVIEW

**Review by aslonnie — COMMENTED** [review]
it will matter if `/tmp/bazel_event_logs` directory is mounted, right? :)

**Comment by can-anyscale** [comment]
@aslonnie yes, it matters in the other PR and is a no-op in this PR.

**Comment by can-anyscale** [comment]
This clean up should probably be done in a job clean up step/post-command hook and not in the life-time of the step

**Comment by aslonnie** [comment]
> The clean up for mac should also be done in a job clean up step/post-command hook and not in the life-time of the step

I totally agree on this! should we just do that instead? and just remove this `cleanup` thing in the `upload` script? it is actually weird that after "uploading", the files will just disappear from the filesystem.. (that should be called "ascending")

**Review by aslonnie — COMMENTED** [review]
maybe just remove this `cleanup()` thing and move it to the `macos` epilogue part?

**Comment by can-anyscale** [comment]
I moved all the clean up to pre_command so it happens only in one place

File: .buildkite/hooks/pre-command [hunk 1]
```diff
@@ -8,26 +8,33 @@
...
+  for dir in "$@"; do
+    if [ -d $dir ]; then
+      echo "Cleaning up old artifacts before command."
+      echo "Artifact directory contents before cleanup:"
+      find $dir -print || true
+
+      if [ "$(ls -A $dir)" ]; then
+        echo "Directory not empty, cleaning up..."
+        # Need to run in docker to avoid permission issues
+        docker run --rm -v $dir:/artifact-mount alpine:latest /bin/sh -c 'rm -rf /artifact-mount/*; rm -rf /artifact-mount/.[!.]*' || true
```
Comment (aslonnie) [inline .buildkite/hooks/pre-command:22]: use `/bin/bash`, for consistency? `/bin/sh` often has a different set of quirks.

File: .buildkite/hooks/pre-command [hunk 2]
```diff
@@ -8,26 +8,33 @@
...
+      else
+        echo "Directory already empty, no need to clean up."
+      fi
 
-    echo "Artifact directory contents after cleanup:"
-    find /tmp/artifacts -print || true
-fi
+      echo "Artifact directory contents after cleanup:"
+      find $dir -print || true
```
Comment (aslonnie) [inline .buildkite/hooks/pre-command:28]: `"$dir"`

**Review by aslonnie — COMMENTED** [review]
high-level, this is `pre-command`, not `post-command`?

we want the clean up to be in `post-command` ?

also does the `docker run` thing also work on osx machine?

**Comment by can-anyscale** [comment]
oh yeah hmm this script doesn't work on macs or windows

**Comment by can-anyscale** [comment]
- Remove the usage of docker, run well on CI
- The clean up can happen safely either on pre or post-command; I use pre-command since that is where the clean up is currently. post-command is currently used for buildkite annotation.

File: .buildkite/hooks/pre-command [hunk 3]
```diff
@@ -8,26 +8,35 @@
...
-    if [ "$(ls -A /tmp/artifacts)" ]; then
-      echo "Directory not empty, cleaning up..."
-      # Need to run in docker to avoid permission issues
-      docker run --rm -v /tmp/artifacts:/artifact-mount alpine:latest /bin/sh -c 'rm -rf /artifact-mount/*; rm -rf /artifact-mount/.[!.]*' || true
-    else
-      echo "Directory already empty, no need to clean up."
-    fi
+_cleanup() {
+  for dir in "$@"; do
+    if [[ -d $dir ]]; then
```
Comment (aslonnie) [inline .buildkite/hooks/pre-command:14]: nit `"$dir"`, double quote.

File: .buildkite/hooks/pre-command [hunk 4]
```diff
@@ -8,26 +8,35 @@
...
+      echo "Cleaning up old artifacts before command."
+      echo "Artifact directory contents before cleanup:"
+      find "$dir" -print || true
+
+      if [[ -z "$($ls -A "$dir")" ]]; then
```
Comment (aslonnie) [inline .buildkite/hooks/pre-command:19]: you want to drop the `$` in `$ls` ?

File: .buildkite/hooks/post-command [hunk 5]
```diff
@@ -23,3 +24,6 @@ if [ -d "/tmp/artifacts/test-summaries" ] && [ "$(ls -A /tmp/artifacts/test-summ
     echo "--- Cleaning up"
     docker run --rm -v /tmp/artifacts:/artifact-mount alpine:latest /bin/sh -c 'rm -rf /artifact-mount/test-summaries' || true
 fi
+
+# clean up bazel logs
```
Comment (aslonnie) [inline .buildkite/hooks/post-command:28]: one last change: could you add a comment here that says something like: "this only has effect when the bazel test runs in the same environment as the buildkite job commands, and has no effect when the tests run inside another test container" ?

many thanks!

**Review by aslonnie — APPROVED** [review]

**Pull request merged**